### PR TITLE
Added chart for sonarqube

### PR DESCRIPTION
**Pull request by rjkernick**

### CODE REVIEW

File: stable/sonarqube/templates/sonarqube-test.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        # copy bats to tools dir
+        cp -R /usr/local/libexec/ /tools/bats/
+      volumeMounts:
+      - mountPath: /tools
+        name: tools
+  containers:
+    - name: {{ .Release.Name }}-ui-test
+      image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+      command: ["/tools/bats/bats", "-t", "/tests/run.sh"]
+      volumeMounts:
```
Comment (trobert2) [inline stable/sonarqube/templates/sonarqube-test.yaml]: wouldn't it make sense to add volumes for the `data`, `configuration` and `extensions` ?

```
          volumeMounts:
            - mountPath: /opt/sonarqube/conf
              name: sonarqube-conf
            - mountPath: /opt/sonarqube/data
              name: sonarqube-data
            - mountPath: /opt/sonarqube/extensions
              name: sonarqube-extensions
```

and create `PVC`s for them? I would be happy to contribute to this PR if it has traction, or try and make a PR of my own.

File: stable/sonarqube/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+              value: "jdbc:postgresql://{{ template "postgresql.fullname" .}}:5432/{{.Values.postgresql.postgresDatabase}}"
+          livenessProbe:
+            httpGet:
+              path: /
+              port: {{ .Values.service.internalPort }}
+          readinessProbe:
+            httpGet:
+              path: /
+              port: {{ .Values.service.internalPort }}
+          resources:
```
Comment (trobert2) [inline stable/sonarqube/templates/deployment.yaml]: wouldn't it make sense to add volumes for the data, configuration and extensions ?
```
          volumeMounts:
            - mountPath: /opt/sonarqube/conf
              name: sonarqube-conf
            - mountPath: /opt/sonarqube/data
              name: sonarqube-data
            - mountPath: /opt/sonarqube/extensions
              name: sonarqube-extensions
```
and create PVCs for them? I would be happy to contribute to this PR if it has traction, or try and make a PR of my own.

Comment (rjkernick) [reply stable/sonarqube/templates/deployment.yaml]: I wouldn't be opposed to that as long as it is easy to turn on and off

File: stable/sonarqube/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "name" -}}
```
Comment (unguiculus) [inline stable/sonarqube/templates/_helpers.tpl]: We've started namespacing templates. See #1785 and update your PR accordingly.

File: stable/sonarqube/templates/pvc-conf.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) }}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ template "fullname" . }}-conf
+  labels:
+    app: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/sonarqube/templates/pvc-conf.yaml]: `app: {{ template "name" . }}`

File: stable/sonarqube/templates/pvc-data.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) }}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ template "fullname" . }}-data
+  labels:
+    app: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/sonarqube/templates/pvc-data.yaml]: `app: {{ template "name" . }}`

File: stable/sonarqube/templates/pvc-extensions.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) }}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ template "fullname" . }}-extensions
+  labels:
+    app: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/sonarqube/templates/pvc-extensions.yaml]: `app: {{ template "name" . }}`

File: stable/sonarqube/templates/sonarqube-test.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-ui-test-{{ randAlphaNum 5 | lower }}"
```
Comment (unguiculus) [inline stable/sonarqube/templates/sonarqube-test.yaml:4]: Add standard labels.

File: stable/sonarqube/values.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
+# Default values for sonarqube.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+replicaCount: 1
+image:
+  repository: sonarqube
+  tag: 6.5
+service:
+  name: sonarqube
+  type: LoadBalancer
```
Comment (unguiculus) [inline stable/sonarqube/values.yaml:10]: I don't think `LoadBalancer` should be the default.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR. You may ignore the CircleCI failure. It's a bug in the new CI scripts for CircleCI.

File: stable/sonarqube/values.yaml (hunk 8 above)
Comment (trobert2) [reply stable/sonarqube/values.yaml:10]: I prefer `LoadBalancer` myself. There was a similar conversation when merging `sonatype-nexus` (see comments on the `values.yaml` file): https://github.com/kubernetes/charts/pull/1915/files/6b77974124c2781ac5d2ee4b7c2c61f76ee62920#diff-2ad2263f9384cb405ca5a6e750e2b51a

Unless there is a standard for this that I don't know about, could we please go forward with it this way?

File: stable/sonarqube/templates/sonarqube-test.yaml (hunk 7 above)
Comment (rjkernick) [reply stable/sonarqube/templates/sonarqube-test.yaml:4]: Added

**Comment by rjkernick** [comment]
@unguiculus Made the suggested changes. Let me know if there is anything else. I would like to leave the default service to LoadBalancer.

File: stable/sonarqube/values.yaml (hunk 8 above)
Comment (unguiculus) [reply stable/sonarqube/values.yaml:10]: OK, let's keep it then. BTW, I noticed there's an incorrect `app` label on the PVC in the Nexus chart. Care to fix it? https://github.com/kubernetes/charts/blob/master/stable/sonatype-nexus/templates/pvc.yaml#L7

File: stable/sonarqube/templates/test-config.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "sonarqube.fullname" . }}-tests
```
Comment (unguiculus) [inline stable/sonarqube/templates/test-config.yaml:4]: Add standard labels.

Comment (rjkernick) [reply stable/sonarqube/templates/test-config.yaml:4]: Do you have an example of what I am missing here?

Comment (unguiculus) [reply stable/sonarqube/templates/test-config.yaml:4]:
```yaml
labels:
  app: {{ template "sonarqube.name" . }}
  chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
  release: {{ .Release.Name }}
  heritage: {{ .Release.Service }}
```

Comment (rjkernick) [reply stable/sonarqube/templates/test-config.yaml:4]: Added

**Pull request merged**